honeybee_radiance/primitive/geometry/bubble.py

```diff
@@ -0,0 +1,17 @@
+"""Radiance Bubble.
+
+http://radsite.lbl.gov/radiance/refer/ray.html#Bubble
+"""
+from .sphere import Sphere
+
+
+class Bubble(Sphere):
+    """Radiance Bubble.
+
+    A bubble is simply a sphere whose surface normal points inward.
```

Comment:
> though I remember that it matters for the mirror surfaces of the direct reflection recipe. Are there any other cases where it matters?

Yes. There are other cases like `glow` and `light` materials. In general the direction mostly matters when materials illuminate light. They should only illuminate light from the front of the face or otherwise you will get extra light from the back-side or the face.

> On a related note, is it also true that the starting vertex does not matter for Radiance (ie. upper-left, lower right, etc.). If that's just an E+ thing, maybe we only enforce counter-clockwise and not the starting vertex.

The starting vertex does not matter in Radiance but now that we are talking about face order this is also something to keep in mind about faces with holes:

https://discourse.radiance-online.org/t/obj2rad-obj2mesh-polygons-or-triangles/4826/5

Reply:
Thanks for shedding some light on this. Now that I understand it a bit more, I can get behind the idea of enforcing counter-clockwise vertices across all Face3Ds in Honeybee.  I don't think it's necessary to enforce it across all of `ladybug_geometry` since this could create inefficiency in places we don't need the check and I don't think we really need to enforce the upper-left vertex until we are exporting to energy simulation (at which point we can just call the `upper_left_counter_clockwise_vertices` property). I make this recommendation because enforcing clockwise order is fairly easy through all of the transformations / edits we might do to the geometry but enforcing the upper left vertex can become hellish through transformations like generating sub-faces, reflections, etc.

However, enforcing counter-clockwise vertices should be pretty efficient and we would just have to add the following code every time that we create a new Face3D in Honeybee:
`face = face.reverse() if face.is_clockwise else face`
Using the `reverse` method will preserve all of the attributes that have already been calculated for the Face3D in order to determine whether it is counter-clockwise (like the Polygon2D, the area, etc).

BTW, with regards to the point that Nathaniel raises in that Radiance forum post, we are all good to go as we already follow his suggestion.  Calling the `vertices` property on a shape with a hole will ensure that the winding order of the holes is opposite that of the boundary. In fact, I am really surprised that Radiance permits the inverse of this but, in any case, we don't need to worry about it.

Reply:
> we would just have to add the following code every time that we create a new Face3D in Honeybee: `face = face.reverse() if face.is_clockwise else face`

We can do this but it looks to be useful enough to two new property to Face that gives you the anti_clockwise and clockwise `face.clockwise_face` and `face.anticlockwise_face`.

Reply:
Sounds good but it seems a little weird to me to have a Face3D be a property of a Face3D. Could we implement it as a method `face.counterclockwise_face()` instead?

Also, FYI, I have been following the convention of American English throughout the libraries (ie. spelling "Color" as Color and not Colour). What this means for the geometry library is that I use the term "counterclockwise" instead of "anticlockwise" (https://en.wikipedia.org/wiki/Clockwise)

Reply:
both sounds good to me.

Reply:
Ok, this is ready:
https://github.com/ladybug-tools/ladybug-geometry/pull/24